katdal/sensordata.py

```diff
@@ -652,10 +652,12 @@ def __str__(self):
             names = sorted([key for key in self.keys()])
             maxlen = max([len(name) for name in names])
             objects = [self.get(name, extract=False) for name in names]
-        obj_reprs = [(("<numpy.ndarray shape=%s type=%s at 0x%x>" % (obj.shape, obj.dtype, id(obj)))
-                     if isinstance(obj, np.ndarray) else repr(obj)) for obj in objects]
-        actual = ['%s : %s' % (str(name).ljust(maxlen), obj_repr) for name, obj_repr in zip(names, obj_reprs)]
-        virtual = ['%s : <function %s.%s>' % (str(pat).ljust(maxlen), func.__module__, func.__name__)
+        obj_reprs = [(f"<numpy.ndarray shape={obj.shape} type={obj.dtype} at 0x{id(obj):x}>"
+                      if isinstance(obj, np.ndarray) else repr(obj)) for obj in objects]
+        actual = ['{} : {}'.format(str(name).ljust(maxlen), obj_repr)
+                  for name, obj_repr in zip(names, obj_reprs)]
+        virtual = ['{} : <function {}.{}>'
+                   .format(str(pat).ljust(maxlen), func.__module__, func.__name__)
```

Comment:
Can be an f-string as shown above.

Reply:
Done, using same tricks as line above it.